mettle/api/api.go

```diff
@@ -72,17 +69,29 @@ func serve(opts grpcutil.Opts, name, requestQueue, responseQueue, preResponseQue
 		preResponses: common.MustOpenTopic(preResponseQueue),
 		jobs:         map[string]*job{},
 	}
+
 	go srv.Receive()
+	conn, err := grpcutil.Dial(fmt.Sprintf("%s:%d", opts.Host, opts.Port), true, opts.CertFile, opts.TokenFile)
```

Comment:
Consider moving nearly all your new code into `getExecutions`; that will make this function a bit neater since it's all logically connected, and will clean up your error handling.

Reply:
Thanks. I'm not sure whether we should add conTLS to the grpcutl.opts? Doesn't seem like it belongs there but also adding an extra param to getExecutions seems unnecessary.

Reply:
yeah `grpcutil.Opts` is very much the server side options. There isn't an equivalent for client options (maybe there should be but not worth doing in this PR)